devlib/native/macosx_native.cpp

```diff
@@ -124,6 +124,45 @@ namespace macxutil {
         free(buffer); // If we failed
         return NULL;
     }
+
+
+    auto convertHexQStringNumberToDecQStringNumber(const QString & hexNumber) -> QString
+    {
+        return QString::number(hexNumber.toInt(nullptr, 16));
+    }
+
+
+    auto extractBusNumberFromLocationId(const QString & locationID) -> QString
+    {
+        return convertHexQStringNumberToDecQStringNumber(locationID.left(2));
+    }
+
+
+    auto extractUsbPortsFromLocationId(const QString & locationID) -> QStringList
+    {
+        auto usbPortsAsHexQStringNumbers = QString{locationID}.remove(0,2) // remove leading bus number
+                                                              .remove("0") // remove trailing zeros
+                                                              .split("", QString::SkipEmptyParts);
+        auto usbPorts = QStringList{};
+        std::transform(usbPortsAsHexQStringNumbers.cbegin(),
+                       usbPortsAsHexQStringNumbers.cend(),
+                       std::back_inserter(usbPorts),
+                       convertHexQStringNumberToDecQStringNumber);
+        return usbPorts;
+    }
```

Comment:
I think that since the scope of these functions and variables is very small there's no need for such explicit names, IMO `QStringNumbers` part is redundant, so I'd rather use names like these:
- `usbPortsAsHex`, `usbPortsInHex` or `hexUsbPorts` - here you've just created it from `QString` and it's local;
- `convertHexStringToDev` or `hexStringToDec`.

---

Nice use of `std::transform` btw, in the bright future it would look like this:
```c++
return hexUsbPorts | std::ranges::views::transform(hexStringToDec);
```

Reply:
* `usbPortsInHex` -> ok
* `convertHexStringToDec`, `hexStringToDec`: 
    * I'm afraid that these names sound like a conversion from `String` to `Int`. Without trailing `QString` we have to look at function declaration every time to understand, that return type is `QString`.
    * Also, this file has different types of strings: `QString`, `CFString`, `UTF8String(char *)`. I wanted to accent that this function only works with `QStrings`.

    To be honest, I tried to remove some parts of this function name, but couldn't. I think `convertHexQStringToDecQString` is my maximum.
@smirnov1993 , what do you think about `convertHexQStringNumberToDecQStringNumber` function name?

Reply:
`convertHexQStringToDecQString` is explicit and verbose, so I vote for this option.

Reply:
Renamed to `convertHexQStringToDecQString`

Reply:
Seems reasonable, I'm resolving this conversation